### PR TITLE
KubeRay Operator add-on

### PR DESCRIPTION
**Pull request by freschri**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/addons/kuberay-operator.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# KubeRay Operator Add-on
```
Comment (elamaran11) [inline docs/addons/kuberay-operator.md:1]: Please make sure the doc index and mkdocs is updated

File: lib/addons/index.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export * from './helm-addon';
 export * from './karpenter';
 export * from './kube-proxy';
 export * from './kube-state-metrics';
```
Comment (elamaran11) [inline lib/addons/index.ts:30]: this addon to be updated to example blueprint stack.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri Nice work but i have few questions:
1. Where is the ask from Partner side to build this addon though this will be an AWSome addition to Blueprints?
2. Are you planning to build a complete pattern to show usage of this KubeRay with a acomplete workload. We can also build a pattern with observability for this? If so please create a issue in patterns and observability repo
3. Update to doc index and mkdocs is missing too

**Comment by freschri** [comment]
> @freschri Nice work but i have few questions:
> 
>     1. Where is the ask from Partner side to build this addon though this will be an AWSome addition to Blueprints?
> 
>     2. Are you planning to build a complete pattern to show usage of this KubeRay with a acomplete workload. We can also build a pattern with observability for this? If so please create a issue in patterns and observability repo
> 
>     3. Update to doc index and mkdocs is missing too

the idea is to build the cdk equivalent of DoEKS's JARK stack, also presented here: https://aws.amazon.com/blogs/containers/deploy-generative-ai-models-on-amazon-eks/

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri Doc index is still not updated so this wont list in list of addons page. Also can you make sure you test the addon with blueprint-construction running it whole and let us know if it works. We can then run e2e

**Review by elamaran11 — APPROVED** [review]
LGTM. @shapirov103 Please check from your end.

File: docs/addons/kuberay-operator.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+This add-on installs the KubeRay Operator [Helm chart](https://github.com/ray-project/kuberay/blob/master/helm-chart/kuberay-operator/README.md).
+
+## Usage
+
+```typescript
+import { Construct } from 'constructs';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+export class DatadogConstruct extends Construct {
```
[the remark this answers is not in the document]
Comment (freschri) [reply docs/addons/kuberay-operator.md]: sorry copy paste error. done.

File: lib/addons/kuberay/index.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { Construct } from 'constructs';
+import merge from "ts-deepmerge";
+import { ClusterInfo, Values } from "../../spi";
+import { createNamespace } from "../../utils";
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from "../helm-addon";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface KubeRayAddOnProps extends HelmAddOnUserProps {
```
[the remark this answers is not in the document]
Comment (freschri) [reply lib/addons/kuberay/index.ts:10]: there are no config values, please see here: https://ray-project.github.io/kuberay/deploy/helm/

File: lib/addons/kuberay/index.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & KubeRayAddOnProps = {
+  name: "kuberay-operator",
+  chart: "kuberay-operator",
+  namespace: "default",
+  version: "1.0.0-rc.0",
```
Comment (shapirov103) [inline lib/addons/kuberay/index.ts:24]: Why use an RC version for the release? I see the have 1.0.0 - let's use that. We generally avoid using non-GA versions as default for the addon (with rare exceptions).

**Review by shapirov103 — CHANGES_REQUESTED** [review]
@freschri please see my comment. we are targetting 1.13 release this week, there is a chance this addon can make it in.

**Comment by shapirov103** [comment]
@freschri please igore the markdown broken links in the above check. We are addressing in a separate PR.

**Comment by elamaran11** [comment]
@freschri Are you planning to close this PR comments?

**Comment by shapirov103** [comment]
@freschri let's address minor feedback and please push something to retrigger the GH actions. I suppose the MD check that is failing should succeed now.

**Comment by freschri** [comment]
> @freschri let's address minor feedback and please push something to retrigger the GH actions. I suppose the MD check that is failing should succeed now.

done

**Comment by shapirov103** [comment]
@freschri I am getting failure on stack destroy consistently, hence it cannot pass e2e. Have you tried dropping (deleting) the blueprint stack with this addon in place?

**Comment by shapirov103** [comment]
consistently leaves one ENI behind in a secondary subnet. this will have to be resolved before we include it in the release. @freschri

**Comment by freschri** [comment]
> consistently leaves one ENI behind in a secondary subnet. this will have to be resolved before we include it in the release. @freschri

addressed

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**